bokehjs/test/unit/core/bokeh_events.ts

```diff
@@ -61,3 +64,88 @@ describe("BokehEvent", () => {
     expect(event.values).to.be.equal({target: widget, action: "do_that"})
   })
 })
+
+describe("AxisClick event", () => {
+  for (const {side, pt} of [{side: "left", pt: xy(5, 200)}, {side: "right", pt: xy(395, 200)}, {side: "above", pt: xy(200, 5)}, {side: "below", pt: xy(200, 395)}]) {
+    it(`should support linear axes on side: ${side}`, async () => {
+      const plot = new Plot({
+        width: 400,
+        height: 400,
+        title: null,
+        toolbar_location: null,
+        x_range: new Range1d({start: 0, end: 10}),
+        y_range: new Range1d({start: 0, end: 10}),
+      })
+      const axis = new LinearAxis()
+      plot.add_layout(axis, side as any)
+
+      const events: AxisClick[] = []
+      axis.on_event(AxisClick, (event: any) => events.push(event))
```

Comment:
The example I suggested to use as a starting point doesn't contain `: any`. This is because the handler in `on_event()` is statically typed based on the event type or even event name.